esmvalcore/future/config/_config_object.py

```diff
@@ -0,0 +1,189 @@
+"""Importable config object."""
+
+import os
+from datetime import datetime
+from pathlib import Path
+
+import yaml
+
+import esmvalcore
+
+from ._config_validators import _validators
+from ._validated_config import ValidatedConfig
+
+
+class ESMValCoreConfig(ValidatedConfig):
```

Comment:
I'm not sure we should use the term "Core" in these user facing objects. How about ESMValToolConfig, or perhaps even ESMValConfig? :-)

Reply:
I'm struggling a lot with the naming conventions, because I find them a bit inconsistent at the moment. I think that if we import something from ESMValCore, it should be named ESMValCoreSomething. Also for better decoupling, because ESMValTool also exists. `ESMValConfig` is a good compromise that avoids the problem altogether.

Reply:
I think I avoided using Config to avoid naming conflicts with other things called (some variation of) config.

Reply:
What about `UserConfig`?

Reply:
I like `Config` as suggested by @Peter9192 , if someone as a naming conflict they can always write
```python
from esmvalcore import Config as ESMValCoreConfig
```
to avoid it